lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -41,6 +43,39 @@ static void EmitNullPropagation(Js::RegSlot targetObjectSlot, ByteCodeGenerator
     );
 }
 
+/// <summary>
+/// The whole optional-chain expression will be wrapped in a UniNode with `knopOptChain`.
+/// Use this function to emit the whole expression.
+/// </summary>
+template <class TEmitProc>
+static void EmitOptionalChainWrapper(ParseNodeUni *pnodeOptChain, ByteCodeGenerator *byteCodeGenerator, FuncInfo *funcInfo, TEmitProc emitChainContent) {
+    Assert(knopOptChain == pnodeOptChain->nop);
+
+    Js::ByteCodeLabel previousSkipLabel = funcInfo->currentOptionalChainSkipLabel;
+
+    // Create a label that can skip the whole chain and store it in `funcInfo`
+    Js::ByteCodeLabel skipLabel = byteCodeGenerator->Writer()->DefineLabel();
+    funcInfo->currentOptionalChainSkipLabel = skipLabel;
+
+    // Acquire slot for the result value
+    // Prefill it with `undefined` (Fallback for short-circuiting)
+    Js::RegSlot resultSlot = funcInfo->AcquireLoc(pnodeOptChain);
+    byteCodeGenerator->Writer()->Reg1(Js::OpCode::LdUndef, resultSlot);
+
+    // Copy values from wrapper to inner expression
+    ParseNodePtr innerNode = pnodeOptChain->pnode1;
+    innerNode->isUsed = pnodeOptChain->isUsed;
+    innerNode->location = pnodeOptChain->location;
```

Comment:
I now use the same location for the wrapper-node and the (inner) expression node.
- This saves me a copy
https://github.com/chakra-core/ChakraCore/blob/6e7d93508ae687b25c88f81e89a2e4bc721c445e/lib/Runtime/ByteCode/ByteCodeEmitter.cpp#L11654-L11657
- And I don't run into issues with `ReleaseLoc` during function-emission
https://github.com/chakra-core/ChakraCore/blob/6e7d93508ae687b25c88f81e89a2e4bc721c445e/lib/Runtime/ByteCode/ByteCodeEmitter.cpp#L8084